lib/features/parkings_view/widgets/parking_wide_tile_card.dart

```diff
@@ -143,3 +131,69 @@ class _RightColumn extends StatelessWidget {
     );
   }
 }
+
+class _FavouriteWidget extends StatefulWidget {
+  const _FavouriteWidget({required this.parking});
+  final Parking parking;
+
+  @override
+  State<_FavouriteWidget> createState() => _FavouriteWidgetState();
+}
+
+class _FavouriteWidgetState extends State<_FavouriteWidget> {
+  bool _isFavourite = false;
+
+  Future<bool> _loadPreference() async {
+    final prefs = await SharedPreferences.getInstance();
+    return prefs.getBool(widget.parking.id) ?? false;
+  }
+
+  Future<void> _updatePreference() async {
+    setState(() {
+      _isFavourite = !_isFavourite;
+    });
+    final prefs = await SharedPreferences.getInstance();
+    await prefs.setBool(widget.parking.id, _isFavourite);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return GestureDetector(
+      onTap: _updatePreference,
+      child: FutureBuilder(
+        future: _loadPreference(),
```

Comment:
So yeah, this is actually funny case here. The linter has problem with it, even though there's nothing wrong with it. Easy fix is to just use `Future.microtask` like this:
```dart
future: Future.microtask(_loadPreference),
```
and linter won't make any more problems. So this linter rule is usefull in many cases but here it actually raises warning without a greater reason I think so you can alway just shut it off with a "disable comment"


## Alternative
So alternatively we can just use riverpod and make a global shared prefs singleton and we can use prefs using `ref.watch` instead of `FutureBuilder`

```dart
@Riverpod(keepAlive: true)
Future<SharedPreferences> sharedPreferencesSingleton(
  SharedPreferencesSingletonRef ref,
) {
  return SharedPreferences.getInstance();
}
```